src/main/java/net/laboulangerie/laboulangeriemmo/api/ability/AbilitiesRegistry.java

```diff
@@ -0,0 +1,74 @@
+package net.laboulangerie.laboulangeriemmo.api.ability;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bukkit.Bukkit;
+
+import net.laboulangerie.laboulangeriemmo.LaBoulangerieMmo;
+
+public class AbilitiesRegistry {
+    private Map<String, Class<? extends AbilityExecutor>> abilities = new HashMap<>();
+    private Map<String, AbilityTrigger> triggerIndex = new HashMap<>();
+
+    public Class<? extends AbilityExecutor> getAbility(String abilityId) {
+        return abilities.get(abilityId);
+    }
+    /**
+     * Add an ability to the AbilitiesRegistry
+     * @param abilityId Unique identifier of the ability
+     * @param abilityClass A class extending {@link net.laboulangerie.laboulangeriemmo.api.ability.AbilityExecutor AbilityExecutor}
+     * @return false if the {@link net.laboulangerie.laboulangeriemmo.api.ability.AbilityArchetype AbilityArchetype}
+     * is invalid (e.g: ability already exists).
+     * Returns true otherwise
+     */
+    public void registerAbility(String abilityId, Class<? extends AbilityExecutor> abilityClass, AbilityTrigger trigger) {
+        if (validateAbility(abilityId)) {
+            abilities.put(abilityId, abilityClass);
+            triggerIndex.put(abilityId, trigger);
+        }
+    }
+    public boolean exists(String abilityId) {
+        return abilities.containsKey(abilityId);
+    }
+    private boolean validateAbility(String abilityId) {
```

Comment:
Peut être ajouter des espaces entre les fonctions pour aéré le code

Reply:
I tend not to when I have very small functions like this